app/app/controllers/caseworker/cbv_flow_invitations_controller.rb

```diff
@@ -5,24 +5,38 @@ class Caseworker::CbvFlowInvitationsController < Caseworker::BaseController
 
   def new
     @site_id = site_id
-    @cbv_flow_invitation = CbvFlowInvitation.new
+    @cbv_flow_invitation = CbvFlowInvitation.new(site_id: site_id)
 
     if @site_id == "ma"
       @cbv_flow_invitation.snap_application_date ||= Date.today
     end
   end
 
   def create
+    invitation_params = base_params.merge(site_specific_params)
+
+    # handle errors from the mail service
     begin
-      invitation_params = base_params.merge(site_specific_params)
-      CbvInvitationService.new.invite(invitation_params, current_user)
-    rescue => ex
+      @cbv_flow_invitation = CbvInvitationService.new.invite(invitation_params, current_user)
+    rescue => e
       flash[:alert] = t(".invite_failed",
                         email_address: cbv_flow_invitation_params[:email_address],
-                        error_message: ex.message
-                       )
-      Rails.logger.error("Error sending CBV invitation: #{ex.class} - #{ex.message}")
-      return redirect_to new_invitation_path(site_id: params[:site_id])
+                        error_message: e.message)
+      redirect_to caseworker_dashboard_path(site_id: params[:site_id])
```

Comment:
Sorry I didn't notice this earlier. We need the `return` here or else the logic below will continue executing and we will get an error with the `render :new` (a "double render" error when you try to tell Rails two different things to do with the response).

```suggestion
      return redirect_to caseworker_dashboard_path(site_id: params[:site_id])
```